### ISSUE TITLE
Provide Better Integration and UI Consistency of AdGuard and AdGuard VPN Regarding DNS Settings

### ISSUE THREAD

**Issue by mahara** [AdguardTeam/AdGuardVPNForWindows#229]
### Issue Details

- **AdGuard VPN version:**
  - 1.2.480.0
- **VPN mode:**
  - VPN
- **Operating system and version:**
  - Windows 10
  - Windows 11
- **Other network-level software:**
  - AdGuard

### Issues

There are currently various DNS settings that are both available and duplicated in both AdGuard and AdGuard VPN.

AdGuard currently provides better DNS server selection and DNS user rules, currently not available in AdGuard VPN.
But when running both apps, AdGuard VPN only uses it own limited DNS settings and ignores AdGuard settings, such as for example DNS server selection and custom DNS user rules.

It would be better to have better integration and consistency between AdGuard and AdGuard VPN regarding DNS settings.

### Expected Behavior

AdGuard VPN should respect DNS settings defined in AdGuard, such as for example DNS server selection and DNS user rules.

OR

AdGuard VPN should provide the same level of UI consistency regarding DNS settings currently available in AdGuard (or simply just copy-paste such features from AdGuard to AdGuard VPN). For example better DNS server selection and custom DNS user rules.

### Actual Behavior

There are current AdGuard VPN DNS settings that are too limited and/or less-customizable compared to DNS settings found in AdGuard.

#### AdGuard

<img width="623" alt="AdGuard - DNS Server Selection" src="https://user-images.githubusercontent.com/238332/160232924-0d13aa6e-2a2f-4aa4-94a9-37598c3bb94b.png">

<img width="768" alt="AdGuard - DNS User Rules" src="https://user-images.githubusercontent.com/238332/160232929-5f40a63b-c5bf-4d8b-b3b6-9f5cbd51f243.png">

#### AdGuard for VPN

<img width="600" alt="AdGuard for VPN - DNS Server Selection" src="https://user-images.githubusercontent.com/238332/160232942-de2e303c-b36c-4fa8-862b-7024aa3269e7.png">

**Comment by rempeskes** [comment]
Who ever Connects with Default at least from my Country dnsleak.com and others show me in Turkey in 8 or 10 different IPs in Turkey as you can understand because of our nature we are not "in love" each other. So i add cloud-flare, google, comodo, openDNS and they leak a lot, all! with AdGuard VPN. At least the last beta fixed the UI crashes in Windows 11.
I hope be fixed soon. Also a lot that helps to choose DNS is the "DNS Jumper" tool, but not with AdGuard VPN.
A last thing while surfing with UK provides only one IP. That results to even pass through Google with Captcha.

Thanks for the effort so time and the people/users reporting such a glitches for VPN!

**Comment by mahara** [comment]
@Aydinv13 But there's an "OR" part here:

> OR
> 
> AdGuard VPN should provide the same level of UI consistency regarding DNS settings currently available in AdGuard (or simply just copy-paste such features from AdGuard to AdGuard VPN). For example, better DNS server selection and custom DNS user rules.

> Nevertheless, we are planning to extend the VPN's settings.

So it's not actually about an overriding-priority issue, but it's more like missing important features in AdGuard VPN for Windows. And also about **duplicated but inconsistent features**.

Hopefully it will make there soon.

Thanks!

**Comment by Aydinv13** [comment]
@mahara Yes, that's what I meant, DNS feature in AdGuard VPN won't be limited
